File: packages/ios/Sources/Components/IconButton/IconButton.figma.swift
import SwiftUI
import Figma

struct KozmosIconButtonConnect: FigmaConnect {
    let component = KozmosIconButton.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=77-1203"

    @FigmaEnum(
        "Variant",
        mapping: [
            "Default": KozmosIconButtonVariant.`default`,
            "Destructive": KozmosIconButtonVariant.destructive,
            "Outline": KozmosIconButtonVariant.outline,
            "Secondary": KozmosIconButtonVariant.secondary,
            "Ghost": KozmosIconButtonVariant.ghost,
            "Link": KozmosIconButtonVariant.link,
            "Glass": KozmosIconButtonVariant.glass
        ]
    )
    var iconVariant: KozmosIconButtonVariant = .ghost

    @FigmaEnum(
        "Size",
        mapping: [
            "Default": KozmosIconButtonSize.`default`,
            "Small": KozmosIconButtonSize.sm,
            "Large": KozmosIconButtonSize.lg
        ]
    )
    var size: KozmosIconButtonSize = .default

    @FigmaEnum(
        "State",
        mapping: [
            "Default": "default",
            "Disabled": "disabled",
            "Loading": "loading"
        ]
    )
    var state: String = "default"

    var body: some View {
        KozmosIconButton(
            iconName: "magnifyingglass",
            variant: self.iconVariant,
            size: self.size,
            isDisabled: self.state == "disabled",
            isLoading: self.state == "loading",
            action: {}
        )
    }
}

File: packages/android/src/main/java/com/kozmos/components/IconButton/IconButton.figma.kt
package com.kozmos.components.iconbutton

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Search
import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=77-1203")
class KozmosIconButtonConnect {
    @FigmaProperty(FigmaType.Enum, "Variant")
    val variant: KozmosIconButtonVariant = Figma.mapping(
        "Default" to KozmosIconButtonVariant.Default,
        "Destructive" to KozmosIconButtonVariant.Destructive,
        "Outline" to KozmosIconButtonVariant.Outline,
        "Secondary" to KozmosIconButtonVariant.Secondary,
        "Ghost" to KozmosIconButtonVariant.Ghost,
        "Link" to KozmosIconButtonVariant.Link,
        "Glass" to KozmosIconButtonVariant.Glass
    )

    @FigmaProperty(FigmaType.Enum, "Size")
    val size: KozmosIconButtonSize = Figma.mapping(
        "Default" to KozmosIconButtonSize.Default,
        "Small" to KozmosIconButtonSize.Sm,
        "Large" to KozmosIconButtonSize.Lg
    )

    @FigmaProperty(FigmaType.Enum, "State")
    val state: String = Figma.mapping(
        "Default" to "default",
        "Disabled" to "disabled",
        "Loading" to "loading"
    )

    @Composable
    fun ComponentExample() {
        KozmosIconButton(
            icon = Icons.Default.Search,
            onClick = {},
            contentDescription = "Icon action",
            variant = variant,
            size = size,
            enabled = state != "disabled",
            isLoading = state == "loading"
        )
    }
}
